### ISSUE TITLE
Perceptual Hash Module

### ISSUE THREAD

**Issue by sambux1** [sambux1/perceptographic#7]
We need to make it possible to run many perceptual hashes as the foundation for the perceptographic hash. We need to add a `perceptual` module like we have a `pph` module. Then we need to connect the two for an end-to-end perceptographic hash.

This is entirely a software problem (no crypto component), so it shouldn't be too difficult. It should be a high priority.

This is a list of some of the main perceptual hashes we need to implement (will be updated):
- [x] `phash`
- [x] `PDQ`

**Comment by sambux1** [comment]
I made the `perceptual` parent class and implemented the `phash` algorithm.

**Comment by sambux1** [comment]
I have now implemented both `phash` and `PDQ` as perceptual hashes.

I also created an `Image` class as a compatibility layer for passing images as input to the perceptual hash functions. An `Image` object takes as input a filename and can be converted to a `PIL.Image` to an `opencv` image as needed.

Lastly, I added a `create()` function to the `Perceptual` base class which is used to automatically create many perceptual hash functions from a single interface.
